src/libopenrave/robotmanipulator.cpp

```diff
@@ -1125,22 +1031,24 @@ bool RobotBase::Manipulator::CheckEndEffectorSelfCollision(CollisionReportPtr re
             continue;
         }
 
-        bool bIsAffected = false;
-        for(size_t ijoint = 0; ijoint < probot->GetJoints().size(); ++ijoint) {
-            if( probot->DoesAffect(ijoint,ilink) && !probot->DoesAffect(ijoint,iattlink) ) {
-                bIsAffected = true;
-                if( probot->CheckLinkSelfCollision(ilink,report) ) {
-                    if( !bAllLinkCollisions ) { // if checking all collisions, have to continue
-                        return true;
+        // check all gripper links including
+        // 1. links that are rigidly attached to the end effector
+        // 2. links that are controlled by joints other than arm joints
+        // 3. links that are connected with passive but non-static joints
+        if( bIgnoreManipulatorLinks ) {
+            FOREACHC(itindependentlink,vindependentinks) {
+                if( *itlink != *itindependentlink && (*itindependentlink)->IsEnabled() ) {
```

Comment:
Since result of ``CheckCollision(linkA, linkB)`` and  ``CheckCollision(linkB, linkA)`` should be the same, we can skip one of them by something like ``ilink < (**itindependentlink).GetIndex()`` instead of ``*itlink != *itindependentlink``?

Reply:
in my understanding from `GetIndependentLinks`, `independentlinks` should never include `*itlink` and the check here is "just in case". so i think we don't need to change the logic here.

Reply:
I see, they are mutually exclusive, then sure I think current logic is fine.